geoapi/routes/streetview.py

```diff
@@ -0,0 +1,265 @@
+    'system': fields.String(required=True),
+    'path': fields.String(required=True)
+})
+
+streetview_folder_import = api.model('TapisFolderImport', {
+    'service': fields.Nested(tapis_file),
+    'system_id': fields.String(),
+    'path': fields.String()
+})
+
+streetview_sequence = api.model('StreetviewSequence', {
+    'id': fields.Integer(),
+    'streetview_instance_id': fields.Integer(),
+    'start_date': fields.DateTime(dt_format='rfc822', required=False),
+    'end_date': fields.DateTime(dt_format='rfc822', required=False),
+    'bbox': fields.String(required=False),
+    'sequence_id': fields.String(required=False)
+})
+
+streetview_organization = api.model('StreetviewOrganization', {
+    'id': fields.Integer(required=False),
+    'streetview_id': fields.Integer(required=False),
+    'name': fields.String(),
+    'slug': fields.String(),
+    'key': fields.String()
+})
+
+streetview_instance = api.model('StreetviewInstance', {
+    'id': fields.Integer(),
+    'streetview_id': fields.Integer(),
+    'system_id': fields.String(),
+    'path': fields.String(),
+   'sequences': fields.List(fields.Nested(streetview_sequence), allow_null=True),
+})
+
+streetview_service = api.model('Streetview', {
+    'id': fields.Integer(),
+    'user_id': fields.Integer(),
+    'token': fields.String(),
+    'service': fields.String(),
+    'service_user': fields.String(),
+    'organizations': fields.List(fields.Nested(streetview_organization), allow_null=True),
+    'instances': fields.List(fields.Nested(streetview_instance), allow_null=True)
+})
+
+@api.route('/')
+class StreetviewListing(Resource):
+    @api.doc(id="getStreetviews",
+             description="Get all streetviews for a user")
+    @api.marshal_with(streetview_service, as_list=True)
+    def get(self):
+        u = request.current_user
+        logger.info("Get all streetview objects user:{}".format(u.username))
+        return StreetviewService.list(u)
+
+    @api.doc(id="createStreetview",
+             description="Create streetview for a user")
+    @api.expect(streetview_params)
+    @api.marshal_with(streetview_service)
+    def post(self):
+        u = request.current_user
+        service = api.payload.get('service')
+        logger.info("Create streetview object for user:{} and service:{}".format(u.username, service))
+        return StreetviewService.create(u, api.payload)
+
+
+@api.route('/<int:streetview_id>/')
+class StreetviewResource(Resource):
+    @api.doc(id="getStreetview",
+             description="Get a streetview object")
+    @api.marshal_with(streetview_service)
+    def get(self, streetview_id: int):
+        u = request.current_user
+        logger.info("Get streetview object of id:{} for user:{}".format(streetview_id, u.username))
+        return StreetviewService.get(streetview_id)
+
+    @api.doc(id="deleteStreetview",
+             description="Delete a streetview object")
+    def delete(self, streetview_id: int):
```

Comment:
we should look at ensuring that users can only delete their own items in these routes.  Same would apply, for the GETs.

Reply:
We have a ticket for this so resolving